[ONNX][dynamo_export] Integrate onnx-rewriter optimizer

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123379

Introduces common standard onnx optimization such as constant, if, controlflow folding and pattern rewrites.

BowenBao committed:
[ONNX][dynamo_export] Integrate onnx-rewriter optimizer

[ghstack-poisoned]

thiagocrepaldi commented on torch/onnx/_internal/exporter.py:
```diff
@@ -1232,6 +1232,16 @@ def export(self) -> ONNXProgram:
+            except ImportError:
+                warnings.warn(
+                    "ONNXScript optimizer is not available. Skipping optimization. "
+                    "Please install latest onnxscript to enable post export optimization."
```
```suggestion
                    "Please `pip install onnxscript` to enable post-export optimization."
```

thiagocrepaldi replied:
giving the users a "copy&paste" solution :)

BowenBao committed:
Address comment; Update dort as well on "[ONNX][dynamo_export] Integrate onnx-rewriter optimizer"


Introduces common standard onnx optimization such as constant, if, controlflow folding and pattern rewrites.


[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge

BowenBao committed:
linter on "[ONNX][dynamo_export] Integrate onnx-rewriter optimizer"


Introduces common standard onnx optimization such as constant, if, controlflow folding and pattern rewrites.


[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge

BowenBao committed:
linter again on "[ONNX][dynamo_export] Integrate onnx-rewriter optimizer"


Introduces common standard onnx optimization such as constant, if, controlflow folding and pattern rewrites.


[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge

titaiwangms commented on torch/onnx/_internal/onnxruntime.py:
```diff
@@ -922,6 +922,20 @@ def maybe_map_to_meta_val(value):
+            try:
+                from onnxscript import optimizer  # type: ignore[import]
+                from onnxscript.rewriter import (  # type: ignore[import]
+                    onnxruntime as ort_rewriter,  # type: ignore[import]
```
Why do we only have onnxruntime rewriter here?
